docs/internal/proposals/error-handling/error-handling.md

```diff
@@ -0,0 +1,54 @@
+# Error handling in Director graphql
+
+In Compass Director, we have to filter errors which should be visible to user and add information which can be readable by machine.
+
+## Error types
+Basic list of errors which should be displayed to the user:
+* InternalError
+* NotFound
+* NotUnique
+* InvalidData
+* InsufficientScopes
+* ConstraintViolation
+* TenantIsRequired
+* TenantNotFound
+
+This list can be split into 2 groups due to different handling:
+* internal errors
+* the rest of errors, which will be presented to the user
+
+In group of internal errors, are errors from:
+* external libraries
+* panics
+* postgreSQL, except errors like NotFound, NotUnique
+
+To deal with the errors we introduced custom errors, which contains error codes.
+
+## Custom errors and theirs error codes:
+
+| Error type           | Error code  |                            Description                                                            |
+|----------------------|-------------|---------------------------------------------------------------------------------------------------|
+| InternalError        | 10          | Error which cannot be handled in director                                                         |
+| NotFound             | 20          | Error indicate that given resource cannot be found and further processing is impossible           |
+| NotUnique            | 21          | Error indicates that given resource is not unique                                                 |
+| InvalidData          | 22          | The input data is invalid, error description should be delivered in error message                 | 
+| InsufficientScopes   | 23          | Error which indicate that the client doesn't have sufficient permissions to execute the operation |
+| ConstraintViolation  | 24          | Error which indicate that this operation can't happen because referenced resource not exist       |
```

Comment:
```suggestion
| ConstraintViolation  | 24          | Indicates that this operation can't happen because the referenced resource does not exist.       |
```